pybamm/solvers/c_solvers/idaklu/CasadiSolverOpenMP_solvers.hpp

```diff
@@ -0,0 +1,231 @@
+#ifndef PYBAMM_IDAKLU_CASADI_SOLVER_OPENMP_HPP
+#define PYBAMM_IDAKLU_CASADI_SOLVER_OPENMP_HPP
+
+#include "CasadiSolverOpenMP.hpp"
+#include "casadi_solver.hpp"
+
+class CasadiSolverOpenMP_Dense : public CasadiSolverOpenMP {
```

Comment:
Looks good, this looks a lot better than my mess of if-statements :)

Reply:
could probably generate all these classes with macros, but the effort might might not the worth the reduced # lines.

Reply:
I wasn't entirely happy with the repetition here either. The macro approach worked nicely and has been implemented now, thanks.